### PR TITLE
fix(service-worker): fix LruList 'remove' bug

### PR DESCRIPTION
**Pull request by manda-linda**
remove method not removing url from state.map

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #22768

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by gkalpak — COMMENTED** [review]
Good catch! I suspect this (possibly together with https://github.com/angular/angular/pull/22325#discussion_r169307645) would also fix #22218.
Since #22325 seems to be stale, would you mind setting `previous`/`next` to `null` for remove nodes (just in case :grin:).

**Comment by manda-linda** [comment]
Updated the PR setting node.previous/next to null before deleting. The change in PR #22325 shouldn't be necessary now since it was mostly a bandaid for this bug.

**Review by gkalpak — COMMENTED** [review]
`.pop()` also needs clearing `.next/.previous`.
Or maybe we could make `.pop()` just call `.remove()` and avoid duplication:

```ts
pop(): string|null {
  const url = this.state.tail;
  if (url !== null) {
    this.remove(url);
  }
  return url;
}
```

**Comment by gkalpak** [comment]
BTW, could you also add the following at the [end of the commit message][1]:

```
Fixes #22218
Fixes #22768
```

[1]: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit

**Review by gkalpak — APPROVED** [review]
One nit. LGTM otherwise.

File: packages/service-worker/worker/src/data.ts [hunk 1]
```diff
@@ -207,6 +194,7 @@ class LruList {
     // The next pointer of the node being inserted gets set to the old head, before the head
     // pointer is updated to this node.
     node.next = this.state.head;
+    node.previous = null;
```
[the remark this answers is not in the document]
Comment (manda-linda) [reply packages/service-worker/worker/src/data.ts]: Hmmm, yeah I guess you're right. I ran across this issue earlier where the head had a previous node, but that must have been before the switch to using 'remove'.

**Pull request closed without merging**